lib/mosaic-objects/src/main/java/org/eclipse/mosaic/lib/objects/addressing/CellMessageRoutingBuilder.java

```diff
@@ -173,4 +130,68 @@ public CellMessageRoutingBuilder udp() {
         return protocol(ProtocolType.UDP);
     }
 
+    public CellMessageRoutingBuilder destination(NetworkAddress networkAddress) {
+        Validate.isTrue(!destinationChanged, "Destination was already set!");
+        this.destination = networkAddress;
+        this.destinationChanged = true;
+        return this;
+    }
+
+    public CellMessageRoutingBuilder destination(String receiverName) {
+        return destination(IpResolver.getSingleton().nameToIp(receiverName).getAddress());
+    }
+
+    public CellMessageRoutingBuilder destination(Inet4Address ipAddress) {
+        return destination(new NetworkAddress(ipAddress));
+    }
+
+    public CellMessageRoutingBuilder destination(byte[] ipv4Address) {
+        return destination(new NetworkAddress(ipv4Address));
+    }
+
+    public CellMessageRoutingBuilder broadcast() {
+        return destination(new NetworkAddress(NetworkAddress.BROADCAST_ADDRESS));
+    }
+
+    public CellMessageRoutingBuilder mbs() {
```

Comment:
what if I call `broadcast().topological().mbs()`?

I think you need to add `Valitate.isTrue(routing != DestinationType.CELL_TOPOCAST, "MBS can only be activated for geographical routing.")`

Reply:
Yes that would be correct. I also just saw the case where we call `.destination(xy).mbs().build()`. This would work, as all checks pass, but we would not have the geographical area for the mbs. This is only given if we also add `.geographical(area)` to the builder. Should we allow mbs to have an area parameter or should I check for this case separatley?

Reply:
> Should we allow mbs to have an area parameter or should I check for this case separatley?

You mean, if we should _enforce_ it (you wrote "allow")? I think so. You can check that (and more?) in the `checkNecessaryValues` method at the end.